frontend/src/leaderboard.js

```diff
@@ -1,36 +1,87 @@
-  while (leaderboard.firstChild) {
-    leaderboard.removeChild(leaderboard.firstChild);
+class Leaderboard {
+  constructor() {
+    this.leaderboardDiv = document.getElementById('users-container');
+    this.leaderboardWindow = document.getElementById('leaderboard-window');
+    this.leaderboardToggleButton = document.getElementById('leaderboard-toggle-button');
+
+    this.displayAccount = this.displayAccount.bind(this);
+    this.displayAccounts = this.displayAccounts.bind(this);
+    this.displayAccountBestRun = this.displayAccountBestRun.bind(this);
+    this.createAccountDiv = this.createAccountDiv.bind(this);
+    this.getBestRun = this.getBestRun.bind(this);
+    this.createBestRunElements = this.createBestRunElements.bind(this);
+    this.fetchUsers = this.fetchUsers.bind(this);
+    this.handleLeaderboardToggle = this.handleLeaderboardToggle.bind(this);
   }
-  document.getElementById('leaderboard-window').classList.remove('hidden');
-  fetchUsers();
-});
 
-function fetchUsers () {
-  fetch(ACCOUNTS_URL)
+  fetchUsers() {
+    fetch("http://localhost:3000/accounts")
     .then(res => res.json())
-    .then(displayAccounts);
-}
-function displayAccounts (accounts) {
-  for (const account of accounts) {
-    displayAccount(createAccountDiv(account));
+    .then(this.displayAccounts);
   }
-}
-function displayAccount (accountDiv) {
-  const leaderboardDiv = document.getElementById('leaderboard-window');
-  leaderboardDiv.append(accountDiv);
-}
-function createAccountDiv (account) {
-  const accountDiv = document.createElement('div');
-  const accountH3 = document.createElement('h3');
-  const accountP = document.createElement('p');
 
-  accountH3.textContent = account.username;
-  accountP.textContent = account.email;
-  accountDiv.append(accountH3, accountP);
+  displayAccounts(accounts) {
+    for (const account of accounts) {
+      this.displayAccount(this.createAccountDiv(account));
+      this.displayAccountBestRun(this.createBestRunElements(account));
+    }
+  }
+  displayAccount(accountDiv) {
+    this.leaderboardDiv.append(accountDiv);
+  }
+
+  displayAccountBestRun(bestRunElements){
+    for (const ele of bestRunElements) {
+      this.leaderboardDiv.append(ele);
+    }
+  }
+  
+  createAccountDiv(account) {
+    const accountDiv = document.createElement('div');
+    const accountH3 = document.createElement('h3');
+    accountH3.textContent = account.username;
+    accountDiv.append(accountH3);
+  
+    return accountDiv;
+  }
+  
+  getBestRun(runs) {
+    let bestRun = runs[0];
+    for (const run of runs) {
+      if (run.score > bestRun.score) {
+        bestRun = run;
+      }
+    }
+    return bestRun;
+  }
+  
+  createBestRunElements(account) {
+    const runScoreP = document.createElement('p');
+    const runWordsTyped = document.createElement('p');
+    const runWordsSeen = document.createElement('p');
+    
+    const bestRun = this.getBestRun(account.runs);
```

Comment:
it looks like account.runs is returning undefined here. When the program then goes to getBestRun it errors out.